src/agent/src/image_rpc.rs

```diff
@@ -119,29 +363,48 @@ impl ImageService {
     }
 
     async fn pull_image(&self, req: &image::PullImageRequest) -> Result<String> {
+        env::set_var("OCICRYPT_KEYPROVIDER_CONFIG", KEYPROVIDER_PATH);
+
         let image = req.get_image();
-        let mut cid = req.get_container_id();
+        let mut cid = req.get_container_id().to_string();
 
         if cid.is_empty() {
             let v: Vec<&str> = image.rsplit('/').collect();
             if !v[0].is_empty() {
-                cid = v[0]
+                cid = v[0].replace(":", "_");
             } else {
                 return Err(anyhow!("Invalid image name. {}", image));
             }
         } else {
-            verify_cid(cid)?;
+            verify_cid(&cid)?;
         }
 
         let source_creds = (!req.get_source_creds().is_empty()).then(|| req.get_source_creds());
 
-        // Read the policy path from the agent config
-        let config_policy_path = &AGENT_CONFIG.read().await.container_policy_path;
-        let policy_path = (!config_policy_path.is_empty()).then(|| config_policy_path);
-        info!(sl!(), "Using container policy_path: {:?}...", &policy_path);
+        if Path::new(SKOPEO_PATH).exists() {
+            // Read the policy path from the agent config
+            let config_policy_path = &AGENT_CONFIG.read().await.container_policy_path;
+            let policy_path = (!config_policy_path.is_empty()).then(|| config_policy_path);
+            info!(sl!(), "Using container policy_path: {:?}...", &policy_path);
```

Comment:
I'd remove the ellipsis as it suggests something else will follow (on that log line)... but it doesn't ;)

Reply:
ok, I'll remove this one